equipment-datastore-rdbms/src/main/java/com/telecominfraproject/wlan/equipment/datastore/rdbms/EquipmentDAO.java

```diff
@@ -532,21 +532,21 @@ public Equipment delete(long equipmentId) {
     }
 	
     public PaginationResponse<Equipment> getForCustomer(int customerId, EquipmentType equipmentType,
-            Set<Long> locationIds, List<ColumnAndSort> sortBy, PaginationContext<Equipment> context) {
+            Set<Long> locationIds, String criteria, List<ColumnAndSort> sortBy, PaginationContext<Equipment> context) {
 
         PaginationResponse<Equipment> ret = new PaginationResponse<>();
         ret.setContext(context.clone());
 
         if (ret.getContext().isLastPage()) {
             // no more pages available according to the context
             LOG.debug(
-                    "No more pages available when looking up equipment {} for customer {} locations {} last returned page number {}",
-                    equipmentType, customerId, locationIds, context.getLastReturnedPageNumber());
+                    "No more pages available when looking up equipment {} for customer {} locations {} criteria {} last returned page number {}",
+                    equipmentType, customerId, locationIds, criteria, context.getLastReturnedPageNumber());
             return ret;
         }
 
-        LOG.debug("Looking up equipment {} for customer {} locations {} last returned page number {}", equipmentType,
-                customerId, locationIds, context.getLastReturnedPageNumber());
+        LOG.debug("Looking up equipment {} for customer {} locations {} criteria {} last returned page number {}", equipmentType,
+                customerId, locationIds, criteria, context.getLastReturnedPageNumber());
 
         String query = SQL_GET_BY_CUSTOMER_ID_AND_EQUIPMENT_TYPE;
 
```

Comment:
queryArgs.add(equipmentType.getId()); <- this will throw NPE when equipmentType is null.

To fix this, a query should be constructed differently - when equipmentType is null, it should not be part of the query at all.

Reply:
also, please add a unit test for equipmentType==null